src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAICitizenWander.java

```diff
@@ -45,12 +46,17 @@ public boolean shouldExecute()
             return false;
         }
 
-        Vec3d vec3d;
+        Vec3d vec3d=null;
         if(citizen.getCitizenData().getSaturation() <= 0)
         {
-            vec3d = new Vec3d(citizen.getColony().getBuildingManager().getBestRestaurant(citizen));
+            final BlockPos pos = citizen.getColony().getBuildingManager().getBestRestaurant(citizen);
+            if(pos != null)
+            {
+                vec3d = new Vec3d(pos);
+            }
         }
-        else
+        
+        if(vec3d == null)
         {
             vec3d = RandomPositionGenerator.getLandPos(citizen, 10, 7);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 7 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)